nservicebus/rabbitmq/configuration-api.md

```diff
@@ -14,197 +14,242 @@ To use RabbitMQ as the underlying transport:
+
+An example of these settings being set in a connection string via code:
+
+snippet:rabbitmq-connection-tls
+
+Or configuration:
 
-By default NServiceBus uses the `message-id` property of the AMQP standard to relay the message id. If this header isn't set the transport will throw an exception since NServiceBus needs a message ID in order to perform retries, de-duplication etc. in a safe way. In integration scenarios where the sender is not controlled consider using a custom scheme that extracts the message ID from e.g.a custom header or some data contained in the actual message body. In these cases in a custom strategy by calling:
+snippet:rabbitmq-connection-tls-config
+
+
+## Controlling the message ID strategy
+
+By default, NServiceBus uses the `message-id` property of the AMQP standard to relay the message ID. If this header isn't set, the transport will throw an exception, because NServiceBus needs a message ID in order to perform retries, de-duplication etc. in a safe way. For integration scenarios where the sender is not controlled, consider using a custom scheme that extracts a message ID from a custom header or some data contained in the actual message body. This custom strategy can be configured by calling:
 
 snippet:rabbitmq-config-custom-id-strategy
 
 WARNING: It is extremely important to use a uniquely identifying property of the message in a custom message ID strategy. If the value for a message were to change (for example, if attempting to use `Guid.NewGuid().ToString()`) then message retries would break, as the infrastructure would be unable to determine that it was processing the same message repeatedly.
 
 
-### Getting full control over the broker connection
+## Getting full control over the broker connection (Versions 3 and below)
 
-NOTE: In Versions 4 and above `IManageRabbitMqConnections` is obsolete and there is no option to provide custom connection manager.
-
-The default connection manager that comes with the transport is usually good enough for most users. To control how the connection(s) with the broker is managed implement a custom connection manager by inheriting from `IManageRabbitMqConnections`. This requires that connections be provided for:
+The default connection manager that comes with the transport is usually good enough for most users. To control how the connection(s) with the broker is managed, implement a custom connection manager by inheriting from `IManageRabbitMqConnections`. This requires that connections be provided for:
 
  1. Administrative actions like creating queues and exchanges
  2. Publishing messages to the broker
  3. Consuming messages from the broker
 
-In order for the transport to use the above register it as shown below:
+In order for the transport to use the above, register it as shown below:
 
 snippet:rabbitmq-config-useconnectionmanager
 
+WARNING: In Versions 4 and above, the ability to provide a custom connection manager has been removed.
```

Comment:
I intentionally didn't put it at the top because I it seemed weird to warn about the setting being removed before getting to the text that explains what the settings is. That's why I included the (Versions 3 and below) in the header.